### ISSUE TITLE
[FEAT] Allow sorting on a related table field using supabase dataprovider

### ISSUE THREAD

**Issue by icomad** [pankod/refine#2066]
**Is your feature request related to a problem? Please describe.**
Right now a table can only be sorted on its own fields, it would be useful to allow sorting on a related table field.

**Describe the solution you'd like**
Using the `metaData` arguments we could add to the `getList` method something like this right before line 118
```js
metaData?.sort?.map((item) => {
    query.order(item.field, { foreignTable: item.foreignTable, ascending: item.order === "asc" });
});
```

This would allow the user to pass an object like this to `useTable` to obtain the table correctly sorted
```js
{
    metaData: {
        select: '*, categories(*)'
        sort: [{ 
            field: 'title',
            foreignTable: 'categories',
            order: 'asc' 
        }]
    }
}
```

**Describe alternatives you've considered**
An alternative solutions could be to allow the use of a field `foreignTable` inside `initialSorter`, `permanentSorter` ecc.. and update lines 118-120 of the `getList` method to this
```js
sort?.map((item) => {
    query.order(item.field, { ascending: item.order === "asc", foreignTable: item.foreignTable });
});
```

**Additional context**
In both of these solutions the user would still be required to pass a custom `select` operation to `metaData`, as illustrated above,  since the related table must be present inside the select clause for supabase to not give an error

**Comment by omeraplak** [comment]
Hey @icomad ,
so how about use like this, 

```
setSorter([
     {
         field: "categories.title",
         order: "desc",
     },
]);
```
On the data provider side, the sort field can be split according to `.` and the first part can be evaluated as `foreignTable`. Thus, it also works with the Tables of frameworks such as Ant Design and Material UI.

**Comment by icomad** [comment]
It seems to be a great solution!

**Comment by omeraplak** [comment]
Hey @icomad ,
We added this feature but it doesn't work on the Supabase side. We'll create a PR next week, but you'll have to wait for it to get fixed on Supabase to start using it.

- https://github.com/supabase/postgrest-js/issues/198
- https://github.com/supabase/supabase/discussions/4255

**Comment by omeraplak** [comment]
Hey @icomad ,
We've released `@pankod/refine-supabase@3.27.0` 🎁
